pipelines/batch/src/main/java/org/openmrs/analytics/JdbcFetchHapi.java

```diff
@@ -148,11 +149,12 @@ public FetchRowsJdbcIo(
       this.dataSourceConfig = dataSourceConfig;
       // Note the constraint on `res.res_ver` ensures we only pick the latest version.
       StringBuilder builder =
-          new StringBuilder(
-              "SELECT res.res_id, res.res_type, res.res_updated, res.res_version, res.res_ver, "
-                  + "ver.res_encoding, ver.res_text FROM hfj_resource res, hfj_res_ver ver "
-                  + "WHERE res.res_type = ? AND res.res_id = ver.res_id AND "
-                  + "res.res_ver = ver.res_ver AND res.res_id % ? = ? ");
+              new StringBuilder(
+                      "SELECT res.res_id, hfi.forced_id, res.res_type, res.res_updated, res.res_ver,"
+                              + " ver.res_encoding, ver.res_text FROM hfj_resource res, hfj_res_ver ver,"
+                              + " hfj_forced_id hfi WHERE res.res_type=? AND res.res_id = ver.res_id AND  "
+                              + " res.res_ver = ver.res_ver AND res.res_id % ? = ? AND hfi.resource_pid ="
```

Comment:
I think this breaks for resources that are created with `POST`; have you tested this for cases where a resource does _not_ have a `forced_id`? I think this needs to be a left join, something like: `hfj_res_ver ver LEFT JOIN
hfj_forced_id hfi ON res_id=resource_pid`

Reply:
Thanks @bashir2 for the clarification. I missed handling this scenario. I have updated it the latest changes.